src/hotspot/share/opto/graphKit.cpp

```diff
@@ -619,6 +619,40 @@ void GraphKit::builtin_throw(Deoptimization::DeoptReason reason, Node* arg) {
     }
   }
 
+  if (treat_throw_as_hot && OptimizeImplicitExceptions) {
+    ciInstanceKlass* ex_ciInstKlass = env()->exception_instanceKlass_for_reason(reason);
+    if (ex_ciInstKlass != NULL) {
+      const TypeKlassPtr *ex_type = TypeKlassPtr::make(ex_ciInstKlass);
+      kill_dead_locals();
+      Node* ex_node = new_instance(makecon(ex_type), NULL, NULL, true);
+      set_argument(0, ex_node);
+      ciMethod* init = ex_ciInstKlass->find_method(ciSymbol::make("<init>"), ciSymbol::make("()V"));
+
+      // The following code is modeled after:
+      // DirectCallGenerator* cg = CallGenerator::for_direct_call(init);
+      // JVMState *new_jvms = cg->generate(sync_jvms());
+      // We can't use that code directly because it assumes at various places that we're
+      // at an invoke bytecode which is in genaral not the case for implicit exceptions.
+
+      GraphKit kit(sync_jvms());
+      address target = SharedRuntime::get_resolve_opt_virtual_call_stub();
+
+      CallStaticJavaNode *call = new CallStaticJavaNode(kit.C, TypeFunc::make(init), target, init);
```

Comment:
At the end `<init>()` will call native `fillInStackTrace()` and nothing else:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/Throwable.java#L255

Should we optimize it by inlining it here so that EA can eliminate above Allocation if it does not escape?